data/abilities.ts

```diff
@@ -2360,6 +2360,14 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 			}
 		},
 		onEnd(source) {
+			let nGasActiveElsewhere = false;
+			for (const pokemon of this.getAllActive()) {
+				if (pokemon !== source && pokemon.hasAbility('Neutralizing Gas')) {
```

Comment:
Not sure that `hasAbility` is a good idea here; compare `ignoringAbility`.

It might be worth sharing the code for neutralising gas being active between here and there, as long as you set this ability state to be ending first.

In fact, I think there might be a spurious `-end` bug with an NGas user switching out while under the effect of Gastro Acid, which setting the ability state to be ending first would fix.

Reply:
hasAbility() seems okay to use in this case since it doesn't ignore itself and covers Gastro Acid. The only ways for `End` to run are losing the Ability somehow, switching out, or becoming affected by Gastro Acid. If you check that Pokemon other than this one actually have a functional Neutralizing Gas then it should be fine?

What seems unnecessary here is using `nGasActiveElsewhere`; you can just `return` right away if this line is true.